### PR TITLE
Added generation of delivery excel. Added familyid to connection

### LINKED ISSUE #279: Backend - Add familyId to connections

### LINKED ISSUE #286: Backend - Add functionality for downloading excel sheet of recipients
Need input from Karianne on how the sheet should be designed and which columns should be included.

### PR DESCRIPTION
**Pull request by joffe97**
Fix #286 - Delivery excel
Fix #279 - Add family id to connections

Added endpoint to download excel sheet: /api/admin/excel/delivery/{Location} (e.g. /api/admin/excel/delivery/Nittedal)
Also fixed a mapping bug in the connection entity, where submitter info was gained from the giver data.

### CODE REVIEW

File: GiEnJul/Controllers/AdminController.cs [hunk 1]
```diff
@@ -12,6 +12,10 @@
 using GiEnJul.Exceptions;
 using GiEnJul.Helpers;
 using GiEnJul.Infrastructure;
+using GiEnJul.Utilities;
+using System.Net.Http;
```
Comment (TheseusDeus) [inline GiEnJul/Controllers/AdminController.cs:16]: Is this line necessary? work fine without line 16

File: GiEnJul/Entities/Connection.cs [hunk 2]
```diff
@@ -65,4 +67,5 @@ public Connection(Giver giver, Recipient recipient)
         public string GiverEmail { get; set; }
         public string GiverPhoneNumber { get; set; }
     }
+
```
Comment (TheseusDeus) [inline GiEnJul/Entities/Connection.cs:70]: remove newline

**Review by TheseusDeus — CHANGES_REQUESTED** [review]
No description on either the issues or the PR about how it works, 
this should be present
Some other minor formatting suggestions

File: GiEnJul/Entities/Connection.cs [hunk 3]
```diff
@@ -24,10 +24,11 @@ public Connection(Giver giver, Recipient recipient)
...
             PersonCount = recipient.PersonCount;
             ReceiverLocation = recipient.Location;
+            FamilyId = recipient.FamilyId;
 
-            SubmitterFullName = giver.FullName;
-            SubmitterEmail = giver.Email;
-            SubmitterPhoneNumber = giver.PhoneNumber;
+            SubmitterFullName = recipient.ContactFullName;
+            SubmitterEmail = recipient.ContactEmail;
+            SubmitterPhoneNumber = recipient.ContactPhoneNumber;
```
Comment (Sommerli) [inline GiEnJul/Entities/Connection.cs:31]: Nice catch 👍

File: GiEnJul/Utilities/ExcelGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+using System.Threading.Tasks;
+
+namespace GiEnJul.Utilities
+{
+    public static class ExcelGenerator
+    {
+        public static async Task<XLWorkbook> Generate(IEnumerable<IExcel> excelEntries)
+        {
+            return await Task.Run(() =>
+            {
```
Comment (RonHaland) [inline GiEnJul/Utilities/ExcelGenerator.cs:15]: This probably shouldn't be in a Task.Run

**Review by RonHaland — COMMENTED** [review]
Not sure how much value Task.Run() gives us here. I would just remove it and make the method synchronous.

**Review by TheseusDeus — DISMISSED** [review]

File: GiEnJul/Utilities/ExcelGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+{
+    public static class ExcelGenerator
+    {
+        public static XLWorkbook Generate(IEnumerable<IExcel> excelEntries)
+        {
+            var workbook = new XLWorkbook();
+            var table = new DataTable();
+            foreach (var entry in excelEntries)
+            {
+                foreach (var header in entry.AsOrderedDictionary().Keys)
```
Comment (RonHaland) [inline GiEnJul/Utilities/ExcelGenerator.cs:16]:
```suggestion
            var headers = excelEntries?.FirstOrDefault().AsOrderedDictionary().Keys ?? new List<string>();
            foreach (var header in headers)
```

**Review by RonHaland — APPROVED** [review]

**Pull request merged**